packages/studio/src/components/ComponentTree.tsx

```diff
@@ -60,24 +57,52 @@ export default function ComponentTree(): JSX.Element | null {
     [openIds, setOpenIds]
   );
 
-  const renderNodeCallback = useCallback(
-    (node: NodeModel<ComponentState>, renderParams: RenderParams) => {
-      const { depth, isOpen, hasChild } = renderParams;
-      if (!node.data) {
-        throw new Error(`Node missing data ${JSON.stringify(node, null, 2)}`);
-      }
-      return (
-        <ComponentNode
-          componentState={node.data}
-          depth={depth}
-          isOpen={isOpen}
-          onToggle={onToggle}
-          hasChild={hasChild}
-        />
-      );
-    },
-    [onToggle]
-  );
+  const handleDrop = useDropHandler();
+  const renderNodeCallback = useNodeRenderer();
+  const renderDragPreview = useDragPreview();
+
+  function useNodeRenderer() {
+    const [
+      selectedComponentUUIDs,
+      clearSelectedComponents,
+      setActiveComponentUUID,
+    ] = useStudioStore((store) => {
+      return [
+        store.pages.selectedComponentUUIDs,
+        store.pages.clearSelectedComponents,
+        store.pages.setActiveComponentUUID,
+      ];
+    });
+
+    const renderNodeCallback = useCallback(
+      (node: NodeModel<ComponentState>, renderParams: RenderParams) => {
+        const { depth, isOpen, hasChild, isDragging } = renderParams;
+        if (!node.data) {
+          throw new Error(`Node missing data ${JSON.stringify(node, null, 2)}`);
+        }
+        if (
+          isDragging &&
+          selectedComponentUUIDs.size !== 0 &&
+          !selectedComponentUUIDs.has(node.data.uuid)
+        ) {
+          setActiveComponentUUID(undefined);
+          clearSelectedComponents();
```

Comment:
I think we'd always want to reset the selected components when the active component changes. Can we update the store so it handles that behavior internally? I think it would be easy to forget to update the selected components in the future

Reply:
Done